truelearn/utils/visualisations/_treemap_plotter.py

```diff
@@ -9,46 +9,76 @@
+
+    The color of the rectangle is used to differentiate different knowledge
+    components.
+    """
+
+    def __init__(
+        self,
+        title: str = "Comparison of learner's subjects",
+        xlabel: str = "",
+        ylabel: str = "",
+    ):
+        """Init a treemap plotter.
+
+        Args:
+            title: The default title of the visualization
+            xlabel: The default x label of the visualization
+            ylabel: The default y label of the visualization
+        """
+        super().__init__(title, xlabel, ylabel)
 
     def plot(
         self,
-        content: Union[Knowledge, List[Tuple]],
+        content: Knowledge,
         topics: Optional[Iterable[str]] = None,
         top_n: Optional[int] = None,
-        *,
-        title: str = "Comparison of learner's subjects",
-        x_label: str = "",
-        y_label: str = "",
         history: bool = False,
     ) -> Self:
-        if isinstance(content, Knowledge):
-            content = self._standardise_data(content, history, topics)
-
-        content = content[:top_n]
+        """Plot the graph based on the given data.
 
-        means = [lst[0] for lst in content]
+        Args:
+            content:
+                The Knowledge object to use to plot the visualisation.
+            topics:
+                The list of topics in the learner's knowledge to visualise.
+                If None, all topics are visualised (unless top_n is
+                specified, see below).
+            top_n:
+                The number of topics to visualise. E.g. if top_n is 5, then the
+                top 5 topics ranked by mean will be visualised.
+            history:
+                Whether to utilize history information in the visualisation.
+                If this is set to True, an attribute called history must be
+                present in all knowledge components.
+        """
+        content_dict, _ = self._standardise_data(content, history, topics)[:top_n]
 
-        variances = [lst[1] for lst in content]
-
-        titles = [lst[2] for lst in content]
+        means, variances, titles, *others = list(zip(*content_dict))
 
         if history:
-            timestamps = [lst[3] for lst in content]
+            timestamps = others[0]
             number_of_videos = []
             last_video_watched = []
             for timestamp in timestamps:
                 number_of_videos.append(len(timestamp))
                 last_video_watched.append(timestamp[-1])
         else:
-            number_of_videos = [None for _ in variances]
-            last_video_watched = [None for _ in variances]
+            number_of_videos = last_video_watched = [None] * len(variances)
 
-        self.figure = go.Figure(
+        self.figure.add_trace(
             go.Treemap(
                 labels=titles,
                 values=means,
-                parents=[""] * len(titles),
+                parents=[None] * len(titles),
                 marker_colors=[
```

Comment:
expected type dict got list[str]

Reply:
Based on the example at https://plotly.com/python/treemaps/, it seems that the list should be correct. But I should use a list of empty strings instead of `None`.

Reply:
> Based on the example at https://plotly.com/python/treemaps/, it seems that the list should be correct. But I should use a list of empty strings instead of `None`.

I meant the line 82 had this error, I have reopened

Reply:
I think it's correct. They use this in their example:

```python
fig = go.Figure(go.Treemap(
    labels = labels,
    values = values,
    parents = parents,
    marker_colors = ["pink", "royalblue", "lightgray", "purple", 
                     "cyan", "lightgray", "lightblue", "lightgreen"]
))
```